features/docs/writing_support_code/hooks/README.md

```diff
@@ -0,0 +1,123 @@
+# Cucumber Hooks
+
+Cucumber proposes several hooks to let you specify some code to be executed at
+different stage of the test execution, like before or after the execution of a
+scenario.
+
+Hooks are part of your support code.
+
+They are executed in the following order:
+
+- [AfterConfiguration](#afterconfiguration-and-installplugin)
+- [InstallPlugin](#afterconfiguration-and-installplugin)
+- [BeforeAll](#beforeall-and-afterall)
+  - Per scenario:
+    - [Around](#around)
+    - [Before](#before-and-after)
+      - Per step:
+        - [AfterStep](#afterstep)
+    - [After](#before-and-after)
+- [AfterAll](#beforeall-and-afterall)
+
+You can define as many hooks as you want. If you have several hooks of the same
+types - for example, several `BeforeAll` hooks - they will be all executed once.
+
+## AfterConfiguration and InstallPlugin
+
+[`AfterConfiguration`](#afterconfiguration) and [`InstallPlugin`](#installplugin)
+hooks are dedicated to plugins and are meant to extend Cucumber. For example,
+[`AfterConfiguration`](#afterconfiguration) allows you to dynamically change the
+configuration before the execution of the tests, and [`InstallPlugin`](#installplugin)
+allows to have some code that would have deeper impact on the execution.
+
+### AfterConfiguration
+
+**Note:** this is a legacy hook. You may consider using [`InstallPlugin`](#installplugin) instead.
```

Comment:
Shall we put on here about it being deprecated, or are we doing that elsewhere>?

Reply:
I would not deprecate it yet, just discourage it. At least for the moment.

But maybe I over-think this 🙇‍♂️

Reply:
I'll create a dedicated PR for deprecating it for the next version 👌